contracts/QuestFactory.sol

```diff
@@ -115,7 +113,85 @@ contract QuestFactory is Initializable, OwnableUpgradeable, AccessControlUpgrade
+        IERC20(rewardTokenAddress_).safeTransferFrom(msg.sender, newQuest_, QuestContract(newQuest_).totalTransferAmount());
+        QuestContract(newQuest_).queue();
+    }
+
+    /// @dev Create an erc20 quest, only accounts with the CREATE_QUEST_ROLE can create quests
+    /// @param rewardTokenAddress_ The contract address of the reward token
+    /// @param endTime_ The end time of the quest
+    /// @param startTime_ The start time of the quest
+    /// @param totalParticipants_ The total amount of participants (accounts) the quest will have
+    /// @param rewardAmount_ The reward amount for an erc20 quest
+    /// @param contractType_ Deprecated, it was used when we had 1155 reward support
+    /// @param questId_ The id of the quest
+    /// @return address the quest contract address
+    function createQuest(
+        address rewardTokenAddress_,
+        uint256 endTime_,
+        uint256 startTime_,
+        uint256 totalParticipants_,
+        uint256 rewardAmount_,
+        string memory contractType_,
+        string memory questId_
+    ) external onlyRole(CREATE_QUEST_ROLE) checkQuest(questId_, rewardTokenAddress_) returns (address) {
+        address newQuest = createQuestInternal(
+            rewardTokenAddress_,
+            endTime_,
+            startTime_,
+            totalParticipants_,
+            rewardAmount_,
+            contractType_,
+            questId_
+        );
+
         QuestContract(newQuest).transferOwnership(msg.sender);
+
+        return newQuest;
+    }
+
+
+    /// @dev Create an erc20 quest and start it at the same time. The function will transfer the reward amount to the quest contract
+    /// @dev only accounts with the CREATE_QUEST_ROLE can create quests
+    /// @param rewardTokenAddress_ The contract address of the reward token
+    /// @param endTime_ The end time of the quest
+    /// @param startTime_ The start time of the quest
+    /// @param totalParticipants_ The total amount of participants (accounts) the quest will have
+    /// @param rewardAmount_ The reward amount for an erc20 quest
+    /// @param contractType_ Deprecated, it was used when we had 1155 reward support
+    /// @param questId_ The id of the quest
+    /// @return address the quest contract address
+    function createQuestAndQueue(
+        address rewardTokenAddress_,
+        uint256 endTime_,
+        uint256 startTime_,
+        uint256 totalParticipants_,
+        uint256 rewardAmount_,
+        string memory contractType_,
+        string memory questId_
+    ) external onlyRole(CREATE_QUEST_ROLE) checkQuest(questId_, rewardTokenAddress_) returns (address) {
+        address newQuest = createQuestInternal(
+            rewardTokenAddress_,
+            endTime_,
+            startTime_,
+            totalParticipants_,
+            rewardAmount_,
+            contractType_,
+            questId_
+        );
+
+        transferTokensAndQueueQuest(newQuest, rewardTokenAddress_);
```

Comment:
Should we have a modifier/check to see if the QuestFactory is approved to transfer the amount of reward tokens?